### PR TITLE
Add lgc.rt dialect

### PR DESCRIPTION
**Pull request by LLJJDD**
This is the prerequisite of turning ray tracing SPIR-V Ops into llvm-dialect.

### CODE REVIEW

**Comment by amdvlk-admin** [comment]
8b20335efd21379da1374680e8de29c534212bed Jenkins build error.
/home/amdvlk/jenkins/workspace/vulkan/sanitized-opensource/Github-PR/llpc-github-pr/driver_build/drivers/llvm-project/llvm/include/llvm/ADT/SmallSet.h:19:10: fatal error: llvm/ADT/STLExtras.h: No such file or directory
compilation terminated.
[1273/3080] Building CXX object compiler/llpc/llvm/lib/CodeGen/CMakeFiles/LLVMCodeGen.dir/PostRAHazardRecognizer.cpp.o
[1274/3080] Building CXX object compiler/llpc/llvm/lib/CodeGen/CMakeFiles/LLVMCodeGen.dir/ParallelCG.cpp.o
[1275/3080] Building CXX object compiler/llpc/llvm/lib/CodeGen/CMakeFiles/LLVMCodeGen.dir/ModuloSchedule.cpp.o
[1276/3080] Building CXX object compiler/llpc/llvm/lib/CodeGen/CMakeFiles/LLVMCodeGen.dir/PreISelIntrinsicLowering.cpp.o

**Comment by amdvlk-admin** [comment]
### Test summary for commit 8b20335efd21379da1374680e8de29c534212bed
<details><summary>CTS tests (Failed: 0/211977)</summary><ul><li>Built with version 1.3.3.1</li>
<details open><summary>Ubuntu 22.04, Navi3x</summary><ul>

<li>Passed: 40067/70655 (56.7%)</li>
<li>Failed: 0/70655 (0.0%)</li>
<li>Not Supported: 30588/70655 (43.3%)</li><li>Warnings: 0/70655 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Navi2x</summary><ul>

<li>Passed: 40106/70653 (56.8%)</li>
<li>Failed: 0/70653 (0.0%)</li>
<li>Not Supported: 30547/70653 (43.2%)</li><li>Warnings: 0/70653 (0.0%)</li>
</ul></details>

<details open><summary>Rhel 8.6, Gfx10</summary><ul>

<li>Passed: 36170/70669 (51.2%)</li>
<li>Failed: 0/70669 (0.0%)</li>
<li>Not Supported: 34499/70669 (48.8%)</li><li>Warnings: 0/70669 (0.0%)</li>
</ul></details>


</ul></details>

File: shared/lgcrt/include/lgcrt/LgcRtDialect.h [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+ *
+ *  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ *  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ *  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ *  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ *  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ *FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+ *IN THE SOFTWARE.
+ *
+ ******************************************************************************/
```
Comment (nhaehnle) [inline shared/lgcrt/include/lgcrt/LgcRtDialect.h:24]: Nit: there are lines here with missing spaces after the '*'

File: shared/lgcrt/lib/LgcRtDialect.cpp [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+ *
+ *  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ *  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ *  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ *  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ *  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ *FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+ *IN THE SOFTWARE.
+ *
+ ******************************************************************************/
```
Comment (nhaehnle) [inline shared/lgcrt/lib/LgcRtDialect.cpp:24]: Same here

**Review by nhaehnle — APPROVED** [review]

**Comment by LLJJDD** [comment]
retest this please

**Comment by amdvlk-admin** [comment]
e4d9d1c3cb21eb3aec7a9baaeb736ec9a174f9a3 Jenkins build error.
/home/amdvlk/jenkins/workspace/vulkan/sanitized-opensource/Github-PR/llpc-github-pr/driver_build/drivers/llvm-project/llvm/include/llvm/Transforms/Utils/BasicBlockUtils.h:20:10: fatal error: llvm/ADT/SetVector.h: No such file or directory
 #include  llvm/ADT/SetVector.h 
          ^~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
[1143/3080] Building CXX object compiler/llpc/llvm/lib/CodeGen/CMakeFiles/LLVMCodeGen.dir/BreakFalseDeps.cpp.o
[1144/3080] Building CXX object compiler/llpc/llvm/lib/CodeGen/CMakeFiles/LLVMCodeGen.dir/BranchRelaxation.cpp.o

**Comment by amdvlk-admin** [comment]
### Test summary for commit e4d9d1c3cb21eb3aec7a9baaeb736ec9a174f9a3
<details><summary>CTS tests (Failed: 0/141320)</summary><ul><li>Built with version 1.3.3.1</li>
<details open><summary>Ubuntu 22.04, Navi3x</summary><ul>

<li>Passed: 40065/70667 (56.7%)</li>
<li>Failed: 0/70667 (0.0%)</li>
<li>Not Supported: 30602/70667 (43.3%)</li><li>Warnings: 0/70667 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Navi2x</summary><ul>

<li>Passed: 40106/70653 (56.8%)</li>
<li>Failed: 0/70653 (0.0%)</li>
<li>Not Supported: 30547/70653 (43.2%)</li><li>Warnings: 0/70653 (0.0%)</li>
</ul></details>


</ul></details>

**Review by nhaehnle — APPROVED** [review]

**Pull request merged**